client/src/lib/CKANApi/DataStore/Query.js

```diff
@@ -232,7 +232,7 @@ class Query {
           const edgeMatch = strict ? '' : '%';
           const matchType = match ? 'ILIKE' : 'NOT ILIKE';
 
-          return `"${queryColumn}" ${matchType} '${edgeMatch}${queryTerm}${edgeMatch}'`;
+          return `CAST("${queryColumn}" AS TEXT) ${matchType} '${edgeMatch}${queryTerm}${edgeMatch}'`;
```

Comment:
This can really negatively affect performance. Casts happen for every row and indexes are not adhered to. Can we somehow leverage the fact we get (and store) the column data type from CKAN?

Reply:
Sure, let me have a look tomorrow